### PR TITLE
Handle Null and Promise.resolve(null) in renderMetaTags + renderLinkTags

### PR DESCRIPTION
**Pull request by drewpc**
Fixed handling of `null` and `Promise.resolve(null)` for `renderMetaTags` and `renderLinkTags`.  Added a test harness for `renderMiddleware` which was used to recreate the problem and ensure successful resolution.  Corresponds to issue #789

### CODE REVIEW

File: packages/react-server/core/__tests__/reactMiddlewareSpec.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		});
+
+		afterEach(() => {
+			mockSocket = null;
+		});
+
+		it("don't render meta tags", (finishTest) => {
+			_testFunctions.renderMetaTags(page, mockSocket)
+				.then(() => {
+					expect(mockSocket.toString()).toMatch('');
```
Comment (gigabo) [inline packages/react-server/core/__tests__/reactMiddlewareSpec.js]: Whoa, cool!

**Comment by gigabo** [comment]
LGTM 👍 

Thanks @drewpc!

**Comment by drewpc** [comment]
I forgot to add a 'normal' values check!  I'll do that now.

**Comment by doug-wade** [comment]
lgtm :shipit:

**Comment by drewpc** [comment]
No changes since I submitted the PR.  I saw that there were merge conflicts based on the latest dependency changes in the master branch and resolved the conflicts.

**Comment by gigabo** [comment]
Thanks for jumping on that.

**Pull request merged**